### PR TITLE
Feat/photos pagination

### PR DESCRIPTION
**Pull request by codebynao**
In addition to the pagination for the photos, I fixed a small issue on the login (lastConnection was not updated) and I replaced the `console.error` with `reply.log.error` to use Fastify login directly

### CODE REVIEW

File: src/controllers/photo.ts [hunk 1]
```diff
@@ -24,14 +24,23 @@ type ExtendedFastifyRequest = FastifyRequest<{
...
+}>
 /**
  * CONTROLLER
  */
-const getAll = async (request: ExtendedFastifyRequest, reply: FastifyReply) => {
+const getAll = async (request: QueryFastifyRequest, reply: FastifyReply) => {
   try {
-    reply.send(await PhotoModel.find({}, '-__v').populate('categories', '-__v').populate('user', '-__v').lean())
+    const { limit, page } = request.query
+    const skip: number = (limit * page) / limit
```
Comment (itsabdessalam) [inline src/controllers/photo.ts]: Let's say we're on page 1 and limit is the default one. We should have

```
const skip = (20 * 1) / 20 => // => 1
```
Why not 0 as on first page we only have to limit?

Comment (codebynao) [reply src/controllers/photo.ts]: You're right `limit * (page - 1)` is the correct formula 😅

File: src/controllers/auth.ts [hunk 2]
```diff
@@ -24,12 +24,17 @@ const login = async (request: LoginFastifyRequest, reply: FastifyReply): Promise
...
+    await user.save()
+
+    user = user.toJSON()
     delete user.password
     delete user.isDeactivated
 
     reply.send({ token, user })
   } catch (error) {
-    console.error('error login user: ', error)
+    reply.log.error('error login user: ', error)
```
Comment (itsabdessalam) [inline src/controllers/auth.ts:37]: Well done for using Fastify logging

**Review by itsabdessalam — APPROVED** [review]
LGTM

**Pull request merged**